### ISSUE TITLE
[BUG] SlopeTransformer returns mulit-index pandas dataframe but not the original index labels.

### ISSUE THREAD

**Issue by twobitunicorn** [sktime/sktime#7017]
**Describe the bug**
Applying the `SlopeTransformer` on a multi-index removes all index labels and replaces them with integers.

**To Reproduce**
```python
<Paste your code here>
```

**Expected behavior**
A multi-index dataframe that contains the original index labels..

System:
    python: 3.12.4 (main, Aug 14 2024, 14:42:56) [Clang 15.0.0 (clang-1500.3.9.4)]
executable: /Users/joseph/.pyenv/versions/3.12.4/bin/python
   machine: macOS-14.6.1-arm64-arm-64bit

Python dependencies:
          pip: 24.2
       sktime: 0.32.1
      sklearn: 1.5.1
       skbase: 0.8.2
        numpy: 1.26.4
        scipy: 1.14.0
       pandas: 2.2.2
   matplotlib: 3.9.2
       joblib: 1.4.2
        numba: 0.60.0
  statsmodels: 0.14.2
     pmdarima: 2.0.4
statsforecast: 1.7.6
      tsfresh: None
      tslearn: None
        torch: None
   tensorflow: None
tensorflow_probability: None

**Comment by fkiraly** [comment]
maybe this is a silly question - which dates should the transformed data have? There is in general not a 1:1 mapping of time stamps before to after.

**Comment by fkiraly** [comment]
For the docstring, it is also a bit confusing, this is how I would format it

```

    Parameters
    ----------
    num_intervals : int, optional, default=8
        number of approx equal segments to split the time series into.

    Examples
    --------
    >>> from sktime.transformations.panel.slope import SlopeTransformer
    >>> from sktime.datasets import load_airline
    >>> y = load_airline()
    >>> transformer = SlopeTransformer()
    >>> y_transformed = transformer.fit_transform(y)
    """
```

**Comment by twobitunicorn** [comment]
I would expect the dates sampled at each slope by the duration that it is wast instantiated in.  What is throwing me off is replacing the value of `franz"`with the value of `0` in the `author` index.  if you add a second `org` you also loose all of those labels.

**Comment by twobitunicorn** [comment]
Actual:
<img width="395" alt="image" src="https://github.com/user-attachments/assets/54a91157-3ea5-46ba-8f22-409e394360b4">
Expected:
<img width="446" alt="image" src="https://github.com/user-attachments/assets/53f0124c-d886-4702-a290-0b8b4f1e6497">

**Comment by twobitunicorn** [comment]
You do bring up a good point.  The `date` should correspond to the step in the slope analysis.

**Comment by fkiraly** [comment]
I see. So, the last, right?

I think it is actually ambiguous, so it would be good to give the user a choice here. Sensible choices I can see:

* current behaviour, integer index of the bin starting at 0
* first index
* last index
* period that goes from first index to the last index, although we need to worry about things like entire days etc. Maybe this is too much hassle to implement correctly to include.

As a recipe on how to change, for this the estimator needs to see the index internally.

I would, hence, add `pd-multiindex` to a list in `X_inner_mtype`, or change it entirely to this.
This ensures that we will see this type (multi-indexed `pandas` dataframe) internally.

It will unfortunately break the internal code since that expects a different data type - so one may have to adapt it. Not sure what is best in terms of runtime.

**Comment by twobitunicorn** [comment]
Yes, the last one.  I like the period idea.

**Comment by fkiraly** [comment]
We should give the user a choice of any of the above imo, and we should leave the current behaviour as default, as someone might be using it.

I would suggest to add a parameter `index_treatment` which can have multiple options.

**Comment by twobitunicorn** [comment]
I like that idea.
